src/modules/dataFetching/hooks/useOptionalColumns.tsx

```diff
@@ -9,42 +15,72 @@ export function useOptionalColumns<T extends { id: string }, QueryVariables>({
 }) {
   // All the column definitions that are marked optional
   const optionalColumns = useMemo(
-    () => (columns || []).filter((col) => col.optional),
+    () => (columns || []).filter((col) => !!col.optional),
     [columns]
   );
 
-  // (Internal to this hook) Which optional column defs are currently shown
-  const [currentOptColDefs, setCurrentOptColDefs] = useState<
-    DataColumnDef<T, QueryVariables>[]
-  >(
-    compact(optionalColumns.filter((col) => !col.optional?.defaultHidden) || [])
+  const currentPath = useCurrentPath();
+
+  // Optional columns to show when the page first loads
+  const initialOptionalColumns = useMemo(() => {
+    const storedOptionalCols = currentPath
```

Comment:
I'm looking at the path of what happens if `columns` is not passed, or if the column set doesn't have any optional columns. I think in that case we should skip hitting local storage, and just return early here.

Reply:
I think it would address https://github.com/greenriver/hmis-frontend/pull/1076#issuecomment-2701896505

Reply:
Hm yes you are definitely right we can skip hitting local storage if there are no optional columns.

I think https://github.com/greenriver/hmis-frontend/pull/1076#issuecomment-2701896505 still needs to be addressed separately. Say you are a user working in 2 projects, one with staff assignment and one without. Both projects' enrollment tables have optional columns, so returning early here will not affect either table.